### PR TITLE
SR2020

### PR DESCRIPTION
**Pull request by kierdavis**
Depends on #6.

### CODE REVIEW

File: SR2020/sr2020-markers [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+            true_code = code + base_offset
+            generate(path, true_code, info.size, marker_type, desc)
+
+    basename = os.path.basename(path)
+    zip_path = "sr-{0}-markers-sr{1}.zip".format(basename, SR_YEAR)
+    if os.path.exists(zip_path):
+        os.remove(zip_path)
+
+    pdfs = glob.glob("{0}*.pdf".format(path))
+    check_call(["zip", "--junk-paths", zip_path] + pdfs)
```
Comment (PeterJCLaw) [inline SR2020/sr2020-markers:72]: For future (no need to change this now): you might enjoy the [`zipfile`](https://docs.python.org/3/library/zipfile.html) module.

**Comment by PeterJCLaw** [comment]
Most odd. Historically I had thought that the PDFs generated by `kokimarker` were deterministic (and thus planned to review by checking the code, running it and asserting no diff), but it seems they're not across platforms. They do seem to be stable on a given platform though (if I run the script twice I get identical files out).

**Comment by PeterJCLaw** [comment]
Looks good to me, though we should ensure that the paper size for the token markers matches what we're telling the teams in the docs.

File: SR2020/sr2020-markers (hunk 1 above)
Comment (kierdavis) [reply SR2020/sr2020-markers:72]: Yeah, that would be much tidier. I'm not sure what prompted me to call out to `zip` instead, other than familiarity.

**Comment by kierdavis** [comment]
> Most odd. Historically I had thought that the PDFs generated by `kokimarker` were deterministic (and thus planned to review by checking the code, running it and asserting no diff), but it seems they're not across platforms. They do seem to be stable on a given platform though (if I run the script twice I get identical files out).

Hmm, I'm not convinced I actually committed the right PDFs in 3f5eed9. I did a rebase to merge multiple "regenerate the markers" commits together but maybe I screwed something up. Github is ordering the commits strangely too.

Regardless, I've regenerated them anyway after making the change to render token markers in multiple paper sizes.

**Comment by PeterJCLaw** [comment]
Can I suggest we do a squash to remove files we don't need? Normally I'd be against that for pushed branches, but here I think the size saving could be worth it.

**Comment by PeterJCLaw** [comment]
The code here looks good. I don't have time this evening to review the PDFs. The few I sampled previously seemed fine to my eyeballs (I didn't validate them using libkoki).

**Comment by kierdavis** [comment]
Rebased to squash all the "regen markers" commits into one.

**Comment by RealOrangeOne** [comment]
Code here looks fine, although I can't speak for the tokens which come out of it, although i'm pretty sure those are fine.

Did this code generate the markers which are up on the website? If so I suggest we merge this and make any amendments in a future PR.

**Comment by kierdavis** [comment]
> Did this code generate the markers which are up on the website?

Yes

**Review by PeterJCLaw — APPROVED** [review]
Looks good. I've now additionally run some of the PDFs generated by the script through a webcam (onto a screen) and `pykoki`'s `basic_example`; they seemed fine. I trust that the PDFs committed to the repo came from the script.

**Pull request merged**